dynamallow/exc.py

```diff
@@ -0,0 +1,38 @@
+# --- Schema exceptions ---
+class MarshModelException(Exception):
+    """Base exception for MarshModel problems"""
+
+
+class ValidationError(MarshModelException):
+    """Schema validation failed"""
+    def __init__(self, raw, schema_name, errors, *args, **kwargs):
+        super(ValidationError, self).__init__(*args, **kwargs)
+        self.errors = errors
+        self.raw = raw
+        self.schema_name = schema_name
+
+    def __unicode__(self):
+        return 'Validation failed for data {} with schema {}. Errors: {}'.format(
```

Comment:
While you implemented it, we don't have test coverage on this line, if we did it would fail on py2.6 here.  :)

You need to make these named to keep compatibility with 2.6 (i.e. {0} {1} ...).

After reading this and thinking about it some, I don't think we should include the data in the error message but instead log it to debug.  The error will likely contain enough information (name of field, reason for validation failure, etc) that you won't need the whole document data, and for times you do (assuming you're tracking down a reproducible problem) it's within reason to enable debug logging to see the full document.